integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/websocket/WsBasicTestCase.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.tests.testcases.withapim.websocket;
+
+import io.netty.handler.codec.http.websocketx.WebSocketClientHandshakeException;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.choreo.connect.tests.apim.ApimBaseTest;
+import org.wso2.choreo.connect.tests.apim.ApimResourceProcessor;
+import org.wso2.choreo.connect.tests.apim.utils.StoreUtils;
+import org.wso2.choreo.connect.tests.util.TestConstant;
+import org.wso2.choreo.connect.tests.util.Utils;
+import org.wso2.choreo.connect.tests.util.websocket.WsClient;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class WsBasicTestCase extends ApimBaseTest {
+    private static final String API_CONTEXT = "websocket-basic";
+    private static final String API_VERSION = "1.0.0";
+    private static final String APPLICATION_NAME = "WebSocketBasicApp";
+    private final Map<String, String> requestHeaders = new HashMap<>();
+
+    private String endpointURL;
+
+    @BeforeClass(alwaysRun = true, description = "Create access token and define endpoint URL")
+    void setEnvironment() throws Exception {
+        super.initWithSuperTenant();
+
+        // Get App ID and API ID
+        String applicationId = ApimResourceProcessor.applicationNameToId.get(APPLICATION_NAME);
+
+        String accessToken = StoreUtils.generateUserAccessToken(apimServiceURLHttps, applicationId,
+                user, storeRestClient);
+        requestHeaders.put(TestConstant.AUTHORIZATION_HEADER, "Bearer " + accessToken);
```

Comment:
shall we add a test for invalid/no auth token as well. also a test when x-wso2-disable-security is provided for the ws api.

Reply:
+1

Reply:
looks like I haven't decoded the resource level disable security extensions for asyncAPIs. Created an issue here. https://github.com/wso2/product-microgateway/issues/2708 Will send the test together with the fix